index.js

```diff
@@ -175,6 +209,28 @@ async function decodeTorrentFile (torrent) {
   result.announce = Array.from(new Set(result.announce))
   result.urlList = Array.from(new Set(result.urlList))
 
+  // Process files (simplified to use same logic for v1 and v2)
+  if (isV2 && torrent.info['file tree']) {
+    // Convert v2 file tree to v1-style files array for consistent processing
+    const files = []
+    function processFileTree (tree, currentPath = []) {
+      for (const [name, entry] of Object.entries(tree)) {
+        const fullPath = [...currentPath, name]
+        if (entry.length !== undefined) {
+          files.push({
+            'path.utf-8': fullPath,
+            length: entry.length
+          })
+        } else {
+          processFileTree(entry, fullPath)
+        }
+      }
+    }
+    processFileTree(torrent.info['file tree'])
+    torrent.info.files = files
```

Comment:
Nice. Flattening the tree for easier processing is the right call, but it seems like we may be implicitly making a hybrid torrent here? Adding `files` without `pieces` would look like an (invalid) torrent file to some consumers. Thinking about what we might need to make piece validation easier downstream, and some method for flattening the file tree to get the path, length, and piece root would be good, but I think we may not want to assign it to the v1 files key